Makefile

```diff
@@ -39,11 +35,15 @@ fmt:
 	gofmt -d -s .
 
 .PHONY: check
-check: all test lint
+check: all test end-to-end-test lint
+
+.PHONY: end-to-end-test
+end-to-end-test: docker-image
+	docker run -it nanotube-test
 
 .PHONY: clean
 clean:
-	rm -f nanotube test/sender/sender test/receiver/receiver
+	git clean -Xf
```

Comment:
This may not always be what people would expect. Expectation usually is: remove everything related to the build and test process. This, however, will remove even vscode config.

Reply:
From the doc: -X
Remove only files ignored by Git. This may be useful to rebuild everything from scratch, but keep manually created files. Seems good to me

Reply:
![image](https://user-images.githubusercontent.com/3339062/109484024-3778d380-7a80-11eb-93aa-36ea2e8aba84.png)

Reply:
:man_shrugging:

Reply:
It will also remove all other personal files people use. This would include per-repo ignores that are not in .gitignore. And we probably don't want that.

Reply:
yeah it's removing `.vscode` because well, it's in .gitignore, as I said

Anyway I'll find an alternative way.